packages/core/src/pubsub/pubsub-message.ts

```diff
@@ -17,32 +51,60 @@ export enum MsgType {
   KEEPALIVE,
 }
 
+export const UpdateMessageCodec = co.intersection(
+  [
+    co.strict({
+      typ: co.literal(MsgType.UPDATE),
+      stream: streamIdAsString,
+      tip: cidAsString,
+    }),
+    co.partial({ model: co.optional(streamIdAsString) }),
+  ],
+  'UpdateMessage'
+)
+// Note: co.TypeOf<typeof UpdateMessageCodec> returns {} so we need to define the type explicitly
 export type UpdateMessage = {
   typ: MsgType.UPDATE
   stream: StreamID
   tip: CID
   model?: StreamID
 }
 
-export type QueryMessage = {
-  typ: MsgType.QUERY
-  id: string
-  stream: StreamID
-}
-
-export type ResponseMessage = {
-  typ: MsgType.RESPONSE
-  id: string
-  tips: Map<string, CID>
-}
+export const QueryMessageCodec = co.strict(
+  {
+    typ: co.literal(MsgType.QUERY),
+    id: co.string,
+    stream: streamIdAsString,
+  },
+  'QueryMessage'
+)
+export type QueryMessage = co.TypeOf<typeof QueryMessageCodec>
```

Comment:
@PaulLeCam Is it a style preference of yours to use different names for `QueryMessageCodec` and `QueryMessage` ? I personally find it easier if they both share the same name, say, `QueryMessage`, and I find it curious why do you think the separate names are better.

Reply:
Main reason was to make the difference explicit, but I'll update 👍